common/djangoapps/pipeline_mako/templates/static_content.html

```diff
@@ -93,17 +93,48 @@
     <%doc>
         Loads a studio-frontend page, with the necessary context. Context is expected
         as a dictionary in the body of this tag.
-
-        Dev note: we could also add the locale-injection script in this block
-            -use a better default than the hardcoded 'en'. There should be a setting or something?
-            -lookup (webpack exported) locale-injection script using lang as key
-            -include it as the first script in this block
     </%doc>
     <%
         body = capture(caller.body)
+        body_dict = json.loads(body)
+        locale = body_dict['lang']
+        messages = "{}"
+        localeData = ""
+
+        if locale != 'en':
+            # because en is the default, studio-frontend will have it loaded by default
+            ietf_to_transifex = {
+                'fr': 'fr_CA',
+                'zh-CN': 'zh_CN',
+                'es-419': 'es_419',
+                'ar': 'ar'
+            }
+
+            messages_path = "{base}/studio-frontend/dist/i18n/messages/{locale}.json".format(
+                base=settings.STATIC_ROOT_BASE,
+                locale=ietf_to_transifex[locale]
+            )
+            with open(messages_path) as inputfile:
+                messages = inputfile.read()
+
+            localeData_path = "{base}/studio-frontend/dist/i18n/locale_data/{locale}.js".format(
+                base=settings.STATIC_ROOT_BASE,
+                locale=locale.split('-')[0]
+            )
+            with open(localeData_path) as inputfile:
+                localeData = inputfile.read()
     %>
-    <script type="text/javascript" id='courseContext'>
-      var studioContext = ${ body | n, decode.utf8};
+    <script type="application/json" id="SFE_i18n_messages">
+      ${ messages | n, decode.utf8}
+    </script>
+    <script type="application/json" id="SFE_i18n_localeScript">
+      {
+        "locale": "${ locale | n, decode.utf8}",
+        "data": "${ localeData.replace('\"', '\\\"') | n, decode.utf8}",
```

Comment:
If we don't escape the quotes here, the browser may interpret the (valid javascript) contents of localeData to be javascript, which causes errors. Escaping them still allows for a successful parse by SFE on the other end.